tools/test/config/range_limits_override_invalid/targets.json

```diff
@@ -0,0 +1,9 @@
+{
```

Comment:
If you'd prefer to not have to specify a `test_target` for every config, I think you can just reuse an existing target from Mbed OS. Then you wouldn't have to specify the same `targets.json` for each test.

This test uses the K64F, you can use it as a reference: https://github.com/ARMmbed/mbed-os/tree/master/tools/test/config/invalid_lib_macro_define

Reply:
I ended up running into an issue using the K64F, I get:
```
tools/build_api.py:139: in get_config                                             
    app_config=app_config)                                                        
tools/build_api.py:312: in prepare_toolchain                                      
    config = config or Config(target, src_paths, app_config=app_config)           
tools/config/__init__.py:497: in __init__                                         
    self.app_config_data.get("custom_targets", {}), tgt)                          
tools/targets/__init__.py:150: in generate_py_target                              
    return target(name, total_data)                                               
tools/targets/__init__.py:129: in target                                          
    ), exc)                                                                       
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _   
                                                                                  
exc = NotSupportedException('target K64F has an incomplete target definition',)   
cause = KeyError(u'K64F',)                                                        
```

Could be something else with my setup though.

Reply:
Hmm, seems like that shouldn't be the case, but I could be wrong. If you mimic the above test's setup you still get that error?

Reply:
I changed the "test_target" to "K64F" in range_limits/test_data.json, seems to match the other test setup. I can poke you or Jimmy tomorrow I'm likely derping on something small.